src/test-helpers.tsx

```diff
@@ -25,3 +27,54 @@ export const fakeUser: ExternalUser = Object.freeze({
   username: 'user-1234',
   /* eslint-enable @typescript-eslint/camelcase */
 });
+
+/*
+ * Repeatedly render a component tree using enzyme.shallow() until
+ * finding and rendering TargetComponent.
+ *
+ * This is useful for testing a component wrapped in one or more
+ * HOCs (higher order components).
+ *
+ * The `componentInstance` parameter is a React component instance.
+ * Example: <MyComponent {...props} />
+ *
+ * The `targetComponent` parameter is the React class (or function) that
+ * you want to retrieve from the component tree.
+ */
+type ShallowUntilTargetParams = {
+  componentInstance: React.ReactElement<any>;
+  targetComponent: any;
```

Comment:
Can we do better than `any` here?

Reply:
I tried a bunch of different things and couldn't seem to figure out what would work. I've now found that `Function` works. I'm not sure if that is 100% accurate, but it's definitely better than `any`.

Reply:
mmh, looking at some docs, I'd suggest `JSXElementConstructor<any>` (`any` refers to the props).